tasks/buildah.yaml

```diff
@@ -59,6 +59,34 @@ spec:
         $(params.BUILD_EXTRA_ARGS) --format=$(params.FORMAT) \
         --tls-verify=$(params.TLSVERIFY) --no-cache \
         -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
+      buildah push $(params.IMAGE) oci:/var/lib/containers/oci
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/redhat-appstudio/syft:v0.46.2
+    name: sbom-source
+    args:
+      - dir:$(workspaces.source.path)
+      - --file=$(workspaces.source.path)/sbom-source.json
+      - --output=cyclonedx-json
+  - image: quay.io/redhat-appstudio/syft:v0.46.2
+    name: sbom-image
+    args:
+      - oci-dir:/var/lib/containers/oci
+      - --file=$(workspaces.source.path)/sbom-image.json
+      - --output=cyclonedx-json
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+  - image: $(params.BUILDER_IMAGE)
+    name: inject-sbom
+    resources: {}
+    script: |
+      container=$(buildah from --storage-driver=$(params.STORAGE_DRIVER) --pull-never $(params.IMAGE))
+      buildah --storage-driver=$(params.STORAGE_DRIVER) copy $container $(workspaces.source.path)/sbom-source.json /root/
+      buildah --storage-driver=$(params.STORAGE_DRIVER) copy $container $(workspaces.source.path)/sbom-image.json /root/
```

Comment:
nit: you can set `workingDir: $(workspaces.source.path)` and then copy both files together -> `copy $container sbom-image.json sbom-source.json  /root/` (not tested). Probably it's not bad to add `--add-history` flag.